### PR TITLE
Hide Sourcegraph operator login by default

### PR DESCRIPTION
**Pull request by rafax**
Based on feedback from @malomarrec the Sourcegraph employee login method we add to the default login screen is confusing for our users. This PR changes our login screen UX - when `hideSourcegraphOperatorLogin` experimental feature (added in this PR, to be enabled by default on Cloud) is enabled, Sourcegraph operator login on our sign-in page will be hidden by default, and only shown when `sourcegraph-operator` URL param is present.

The current (suboptimal) check based on provider type/name will be improved when [RFC 744](https://docs.google.com/document/d/1zHW_bBB5xeTqPcgt4UcW9BI-oKvG_jMwooriolsZB-0/edit?usp=sharing) is implemented. 

New default (with flag on):
Screenshot 2022-10-03 at 10 39 43

When flag is off or (when flag is on and with URL param):
Screenshot 2022-10-03 at 10 39 53

## Test plan

- tested locally
- sign-in page unit tests

## App preview:

- [Web](https://sg-web-hide-sourcegraph-login.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ordnskflrs.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by filiphaftek — APPROVED** [review]
Nice 👍

**Review by malomarrec — APPROVED** [review]
Cool!

**Comment by danielmarquespt** [comment]
I think this addresses this issue: https://github.com/sourcegraph/customer/issues/1314 

Right?

**Comment by malomarrec** [comment]
@michaellzc does this close customer#1314?

**Review by kopancek — APPROVED** [review]
LGTM :shipit:

**Review by pjlast — APPROVED** [review]

**Review by unknwon — APPROVED** [review]
😍

File: client/web/src/auth/SignInPage.tsx [hunk 1]
```diff
@@ -35,16 +40,28 @@ export const SignInPage: React.FunctionComponent<React.PropsWithChildren<SignInP
     const location = useLocation()
     const [error, setError] = useState<Error | null>(null)
 
+    const isOperatorHidingEnabled = props.context.experimentalFeatures.hideSourcegraphOperatorLogin ?? false
+
+    const showSourcegraphOperatorLogin =
+        !isOperatorHidingEnabled || new URLSearchParams(location.search).has('sourcegraph-operator')
+
+    const isSourcegraphOperatorProvider = (provider: AuthProvider): boolean =>
+        provider.serviceType === 'openidconnect' && provider.displayName === 'Sourcegraph Employee'
```
Comment (bobheadxi) [inline client/web/src/auth/SignInPage.tsx:49]: Maybe we should take the opportunity to align the naming:

```suggestion
        provider.serviceType === 'openidconnect' && provider.displayName === 'Sourcegraph Operator'
```

Comment (rafax) [reply client/web/src/auth/SignInPage.tsx:49]: We'd need to modify all the names configured in instances + we have other relying on the name, so I'd rather keep this as-is and wait until https://docs.google.com/document/d/1Ia9sjW_KQ6BeJ28xJl3VvLIy7M9Ko7A1sBnTXkZ3W9o/edit#heading=h.trqab8y0kufp is implemented to check by type

**Pull request merged**